synapse/metrics/_gc.py

```diff
@@ -81,6 +83,42 @@ def collect(self) -> Iterable[Metric]:
 if not running_on_pypy:
     REGISTRY.register(GCCounts())
 
+    # disable automatic GC, and replace it with a task that runs every 100ms, so that
+    # we can get some metrics about GC activity.
```

Comment:
If this is just about gathering metrics, could we use the [`gc.callbacks` mechanism](https://docs.python.org/3/library/gc.html#gc.callbacks) instead and not disable automatic gc?

Reply:
per https://github.com/matrix-org/synapse/issues/9813#issuecomment-1009175679: I don't think so.

Reply:
Thanks for answering the same question again. 😢